From fa889f912fe95d4ab4cfeff8c225a8051940a74c Mon Sep 17 00:00:00 2001
From: Tim Murray <timmurray@google.com>
Date: Mon, 15 Oct 2018 16:29:15 -0700
Subject: [PATCH 48/58] hwui: purge malloc pages on bitmap destruction

Immediately purge malloc pages on bitmap destruction. Bitmaps are often
big and can cause memory to stay high for much longer than it should.

Test: boots and works
bug 117795621

Merged-in: If2e8c5f1fc07039cf3dc3edcd3dc06861dbce1a1
Change-Id: If2e8c5f1fc07039cf3dc3edcd3dc06861dbce1a1
(cherry picked from commit 535fae32e597445a480896ea8e01662ada444c0c)
---
 libs/hwui/hwui/Bitmap.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/hwui/hwui/Bitmap.cpp b/libs/hwui/hwui/Bitmap.cpp
index 263d249d20e..c6ef090c1f7 100644
--- a/libs/hwui/hwui/Bitmap.cpp
+++ b/libs/hwui/hwui/Bitmap.cpp
@@ -223,6 +223,7 @@ Bitmap::~Bitmap() {
             break;
         case PixelStorageType::Heap:
             free(mPixelStorage.heap.address);
+            mallopt(M_PURGE, 0);
             break;
         case PixelStorageType::Hardware:
             auto buffer = mPixelStorage.hardware.buffer;
@@ -230,7 +231,6 @@ Bitmap::~Bitmap() {
             mPixelStorage.hardware.buffer = nullptr;
             break;
     }
-
     android::uirenderer::renderthread::RenderProxy::onBitmapDestroyed(getStableID());
 }
 
-- 
2.17.1

